rdm/audit_for_gaps.py

```diff
@@ -175,7 +178,69 @@ def _sorted_checklist_items(unsorted_checklist):
         key = checklist_item.get('reference', '')
         description = checklist_item.get('description', '')
         unsorted_items.append(key + ' ' + description)
-    return sorted(unsorted_items)
+    # The use of the SectionalAnalysis tool ensures '62304:5.1.2' < '62304:5.1.11' < '62304:5.2.1'
+    return sorted(unsorted_items, key=SectionalAnalysis)
```

Comment:
I think it would be cleaner to sort `unsorted_checklist` before concatenating the key and description because then the sorting mechanism can be blissfully unaware of the description.

Also, perhaps we should filter out lines with blank keys up-front? In pseudo code this may go like:

1. filter out blank strings
2. sort the items based only on the references (the key function may have to pull out just the key)
3. concatenate the key and descriptions after

Reply:
What happens if two different checklists use the same key with altered descriptions? The current method happily includes both ordered by description.

Reply:
Ahh! Great point, I had not considered this.

Reply:
Yeah, then your current approach makes a lot of sense.